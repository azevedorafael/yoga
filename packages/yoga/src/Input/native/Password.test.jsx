import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import { ThemeProvider, Input } from '../..';

describe('<Input.Password />', () => {
  describe('Snapshots', () => {
    it('should match with default input', () => {
      const { container } = render(
        <ThemeProvider>
          <Input.Password />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match when input is focused', () => {
      const { container, getByTestId } = render(
        <ThemeProvider>
          <Input.Password label="Input" testID="input" />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));

      expect(container).toMatchSnapshot();
    });

    it('should match with disabled input', () => {
      const { container } = render(
        <ThemeProvider>
          <Input.Password disabled />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with full width', () => {
      const { container } = render(
        <ThemeProvider>
          <Input.Password label="Label" full />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('should call onChangeText', () => {
      const onChangeTextMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input.Password
            label="Input"
            testID="input"
            onChangeText={onChangeTextMock}
          />
        </ThemeProvider>,
      );

      fireEvent.changeText(getByTestId('input'), 'foo');

      expect(onChangeTextMock).toHaveBeenCalled();
    });

    it('should not call onChangeText when input is readOnly', () => {
      const onChangeTextMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input.Password
            label="Input"
            testID="input"
            onChangeText={onChangeTextMock}
            readOnly
          />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));

      expect(onChangeTextMock).not.toHaveBeenCalled();
    });

    it('should not call onChangeText when input is disabled', () => {
      const onChangeTextMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input.Password
            label="Input"
            testID="input"
            onChangeText={onChangeTextMock}
            disabled
          />
        </ThemeProvider>,
      );

      fireEvent.changeText(getByTestId('input'), 'foo');

      expect(onChangeTextMock).not.toHaveBeenCalled();
    });

    it('should call onFocus', () => {
      const onFocusMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input.Password label="Input" testID="input" onFocus={onFocusMock} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));

      expect(onFocusMock).toHaveBeenCalled();
    });

    it('should call onBlur', () => {
      const onBlurMock = jest.fn();

      const { getByTestId } = render(
        <ThemeProvider>
          <Input.Password label="Input" testID="input" onBlur={onBlurMock} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));
      fireEvent.blur(getByTestId('input'));

      expect(onBlurMock).toHaveBeenCalled();
    });
  });

  describe('visibility button', () => {
    it('should toggle password visibility when press visibility button', () => {
      const { getByRole, getByTestId } = render(
        <ThemeProvider>
          <Input.Password label="Input" testID="input" />
        </ThemeProvider>,
      );

      expect(getByTestId('input').props.secureTextEntry).toBe(true);

      fireEvent.press(getByRole('button'));

      expect(getByTestId('input').props.secureTextEntry).toBe(false);
    });
  });
});
