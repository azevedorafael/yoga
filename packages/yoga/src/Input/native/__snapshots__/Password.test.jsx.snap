// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input.Password /> Snapshots should match when input is focused 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={false}
    style={
      Array [
        Object {
          "position": "relative",
          "width": 320,
        },
      ]
    }
  >
    <View
      full={false}
      style={
        Array [
          Object {
            "position": "relative",
            "width": 320,
          },
          Object {},
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCompleteType="password"
        blurOnSubmit={false}
        cleanable={false}
        clearTextOnFocus={false}
        disabled={false}
        editable={true}
        focus={true}
        full={false}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "borderColor": "#41414A",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#41414A",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 40,
              "paddingTop": 16,
              "width": 320,
            },
            Object {
              "height": 52,
            },
          ]
        }
        testID="input"
        textContentType="password"
        typed={false}
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        focus={true}
        pointerEvents="none"
        style={
          Object {
            "left": 14,
            "position": "absolute",
            "top": 24,
            "transform": Array [
              Object {
                "translateY": -8,
              },
            ],
          }
        }
        typed={false}
      >
        <Text
          disabled={false}
          focus={true}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "color": "#41414A",
              "fontSize": 14,
              "fontWeight": "700",
              "paddingLeft": 4,
              "paddingRight": 4,
            }
          }
          typed={false}
        >
          Input
        </Text>
      </View>
    </View>
    <TouchableWithoutFeedback
      accessibilityRole="button"
    >
      <View
        disabled={false}
        style={
          Array [
            Object {
              "paddingLeft": 8,
              "paddingRight": 16,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={52}
          bbWidth={20}
          fill="#41414A"
          focusable={false}
          height={52}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 52,
                "width": 20,
              },
            ]
          }
          width={20}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4282466634,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGSvgView>
      </View>
    </TouchableWithoutFeedback>
  </View>
</View>
`;

exports[`<Input.Password /> Snapshots should match with default input 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={false}
    style={
      Array [
        Object {
          "position": "relative",
          "width": 320,
        },
      ]
    }
  >
    <View
      full={false}
      style={
        Array [
          Object {
            "position": "relative",
            "width": 320,
          },
          Object {},
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCompleteType="password"
        blurOnSubmit={false}
        cleanable={false}
        clearTextOnFocus={false}
        disabled={false}
        editable={true}
        focus={false}
        full={false}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "borderColor": "#D3D3E2",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#9898A7",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 40,
              "paddingTop": 16,
              "width": 320,
            },
            Object {
              "height": 52,
            },
          ]
        }
        textContentType="password"
        typed={false}
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <TouchableWithoutFeedback
      accessibilityRole="button"
    >
      <View
        disabled={false}
        style={
          Array [
            Object {
              "paddingLeft": 8,
              "paddingRight": 16,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={52}
          bbWidth={20}
          fill="#9898A7"
          focusable={false}
          height={52}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 52,
                "width": 20,
              },
            ]
          }
          width={20}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4288190631,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGSvgView>
      </View>
    </TouchableWithoutFeedback>
  </View>
</View>
`;

exports[`<Input.Password /> Snapshots should match with disabled input 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={false}
    style={
      Array [
        Object {
          "position": "relative",
          "width": 320,
        },
      ]
    }
  >
    <View
      full={false}
      style={
        Array [
          Object {
            "position": "relative",
            "width": 320,
          },
          Object {},
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCompleteType="password"
        blurOnSubmit={false}
        cleanable={false}
        clearTextOnFocus={false}
        disabled={true}
        editable={false}
        focus={false}
        full={false}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "borderColor": "#D3D3E2",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#D3D3E2",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 40,
              "paddingTop": 16,
              "width": 320,
            },
            Object {
              "height": 52,
            },
          ]
        }
        textContentType="password"
        typed={false}
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <TouchableWithoutFeedback
      accessibilityRole="button"
    >
      <View
        disabled={true}
        style={
          Array [
            Object {
              "paddingLeft": 8,
              "paddingRight": 16,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={52}
          bbWidth={20}
          fill="#D3D3E2"
          focusable={false}
          height={52}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 52,
                "width": 20,
              },
            ]
          }
          width={20}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4292072418,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGSvgView>
      </View>
    </TouchableWithoutFeedback>
  </View>
</View>
`;

exports[`<Input.Password /> Snapshots should match with full width 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    full={true}
    style={
      Array [
        Object {
          "position": "relative",
          "width": "100%",
        },
      ]
    }
  >
    <View
      full={true}
      style={
        Array [
          Object {
            "position": "relative",
            "width": "100%",
          },
          Object {},
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCompleteType="password"
        blurOnSubmit={false}
        cleanable={false}
        clearTextOnFocus={false}
        disabled={false}
        editable={true}
        focus={false}
        full={true}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "borderColor": "#D3D3E2",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#9898A7",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 40,
              "paddingTop": 16,
              "width": "100%",
            },
            Object {
              "height": 52,
            },
          ]
        }
        textContentType="password"
        typed={false}
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        focus={false}
        pointerEvents="none"
        style={
          Object {
            "left": 16,
            "position": "absolute",
            "top": 24,
            "transform": Array [
              Object {
                "translateY": -8,
              },
            ],
          }
        }
        typed={false}
      >
        <Text
          disabled={false}
          focus={false}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "color": "#9898A7",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
          typed={false}
        >
          Label
        </Text>
      </View>
    </View>
    <TouchableWithoutFeedback
      accessibilityRole="button"
    >
      <View
        disabled={false}
        style={
          Array [
            Object {
              "paddingLeft": 8,
              "paddingRight": 16,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={52}
          bbWidth={20}
          fill="#9898A7"
          focusable={false}
          height={52}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 52,
                "width": 20,
              },
            ]
          }
          width={20}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4288190631,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGSvgView>
      </View>
    </TouchableWithoutFeedback>
  </View>
</View>
`;
